static/app/views/eventsV2/table/quickContext.tsx

```diff
@@ -196,13 +153,113 @@ function IssueContext(props: IssueContextProps) {
   );
 }
 
+type ContextProps = {
+  children: React.ReactNode;
+  contextType: ContextType;
+  dataRow: EventData;
+  organization?: Organization;
+};
+
+export function QuickContextHoverWrapper(props: ContextProps) {
+  const api = useApi();
+  const [ishovering, setisHovering] = useState<boolean>(false);
+  const [error, setError] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [data, setData] = useState<Group | null>(null);
+  const delayOpenTimeoutRef = useRef<number | undefined>(undefined);
+
+  const handleHoverState = () => {
+    setisHovering(prevState => !prevState);
+  };
+
+  const fetchData = () => {
+    if (!data) {
+      const params = getRequestParams(
+        props.dataRow,
+        props.contextType,
+        props.organization
+      );
+      api
+        .requestPromise(params.path, params.options)
+        .then(response => {
+          setData(response);
+          if (isIssueContext(props.contextType)) {
+            GroupStore.add([response]);
+          }
+        })
+        .catch(() => {
+          setError(true);
+        })
+        .finally(() => {
+          setLoading(false);
+        });
+    }
+  };
+
+  const handleMouseEnter = () => {
+    handleHoverState();
+    delayOpenTimeoutRef.current = window.setTimeout(() => {
+      fetchData();
+    }, DATA_FETCH_DELAY);
+  };
```

Comment:
I think I understand why you added this, but can we not rely on the `Hovercard` to render the quick context with a delay, and the quick context itself to fetch its data?

I'm not exactly sure if pulling this `fetchData` logic out to the hover wrapper is necessary

Reply:
Hey Nar. I initially had that.  I did this as a simple approach to the  'fetch only once' logic for optimization, without having to cache.